include/config/source/yaml_file.h

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#include <filesystem>
+#include <utility>
+
+#include <yaml-cpp/yaml.h>
+
+#include "config/config_source_interface.h"
+
+namespace silo::config {
+
+class YamlFile : public ConfigSource {
+   std::string debug_context;
+   std::unordered_map<ConfigKeyPath, YAML::Node> yaml_fields;
+
+   YamlFile(std::string debug_context, std::unordered_map<ConfigKeyPath, YAML::Node> yaml_fields)
+       : debug_context(std::move(debug_context)),
+         yaml_fields(std::move(yaml_fields)) {}
+
+   std::string debugContext() const override;
+
+  public:
+   [[nodiscard]] VerifiedConfigAttributes verify(const ConfigSpecification& config_specification
+   ) const override;
+
+   // Make yaml_fields accessible for testing
+   const std::unordered_map<ConfigKeyPath, YAML::Node>& getYamlFields() const;
+
+   static YamlFile readFile(const std::filesystem::path& path);
+
+   static YamlFile fromYAML(const std::string& debug_context, const std::string& yaml_string);
+
+   static std::string configKeyPathToString(const ConfigKeyPath& key_path);
+
+   static ConfigKeyPath stringToConfigKeyPath(const std::string& key_path_string);
```

Comment:
Alex argued the same and even tried out the change. Reasons why not:

- These methods are different for each config source back-end; each back-end has those methods; the only reason why to add them to ConfigKeyPath is to provide for the "standard" formatting; but it's not clear whether the standard formatting should really be the YAML one, who knows if we add JSON or TOML or whatever in the future and we find another formatting better for debugging? Deciding to explicitly call the Yaml version from debugging info seemed honest and explicit to me.

- Alex argued that the env var etc. methods could instead be moved to ConfigKeyPath, too; but that strikes me odd, to have to change ConfigKeyPath when a new back-end is added, wrong separation of concerns; and this is proven by the fact that the environment variables back-end currently hard-codes the "SILO_" prefix, which prevents the config library from being used unchanged for other apps. We don't need that currently and for that reason it's still hard-coded; but imagine you wanted to remove that hard-coded value (and I think it, besides also specifying part of the help text, might be the only change needed to in fact make the config part independent): the best place to add it would be the `EnvironmentVariables` type; and that means that those 2 methods need to be on `EnvironmentVariables` to get access to that prefix.

Reply:
After I've seen the whole code, I agree. It's better to leave them separate for every config source.